### PR TITLE
Upgrade to Robolectric 3.1

### PR DESCRIPTION
**Pull request by WonderCsabo**
Implements #1375.

This is a WIP, because the changes in `RobolectricTestRunner` in 3.0 killed my workaround. I am waiting on feedback from the Robolectric team.

### CODE REVIEW

File: AndroidAnnotations/functional-test-1-5/src/test/java/org/androidannotations/test15/SeekBarChangeListenedActivityTest.java [hunk 1]
```diff
@@ -38,13 +38,16 @@
 	public void setUp() {
 		activity = setupActivity(SeekBarChangeListenedActivity_.class);
 		seekBar = (SeekBar) activity.findViewById(R.id.seekBar1);
-		shadowSeekBar = shadowOf_(seekBar);
+		shadowSeekBar = (ShadowSeekBar) extract(seekBar);
```
Comment (dodgex) [inline AndroidAnnotations/functional-test-1-5/src/test/java/org/androidannotations/test15/SeekBarChangeListenedActivityTest.java]: it seems that this field is only used inside of `testSeekBarTouchStopNamingConvention` but all other tests get thier own shadow?

Comment (WonderCsabo) [inline AndroidAnnotations/functional-test-1-5/src/test/java/org/androidannotations/test15/SeekBarChangeListenedActivityTest.java]: We need to get the reference to the shadow to access the `getOnSeekBarChangeListener()` method, which is not present in the normal class.

Comment (dodgex) [inline AndroidAnnotations/functional-test-1-5/src/test/java/org/androidannotations/test15/SeekBarChangeListenedActivityTest.java]: but you repeat this call with a local member in each test method but `testSeekBarTouchStopNamingConvention` there you use the field yoou aquire in `setUp`. thats what i meant.

Comment (WonderCsabo) [inline AndroidAnnotations/functional-test-1-5/src/test/java/org/androidannotations/test15/SeekBarChangeListenedActivityTest.java]: Hmm, you are right.

Comment (WonderCsabo) [inline AndroidAnnotations/functional-test-1-5/src/test/java/org/androidannotations/test15/SeekBarChangeListenedActivityTest.java]: PR updated.

**Comment by WonderCsabo** [comment]
I am almost done with this, however i could not create a new workaround for the `ParameterizedRobolectricTestRunner.` I proposed a fix on Robolectric's side https://github.com/robolectric/robolectric/pull/2016. I really hope they will accept it.

**Comment by dodgex** [comment]
@WonderCsabo there is a 3.1 RC.

**Comment by WonderCsabo** [comment]
Thanks for the heads up. :)

**Comment by dodgex** [comment]
@WonderCsabo looks like [3.1](https://github.com/robolectric/robolectric/releases/tag/robolectric-3.1) has been released

**Comment by WonderCsabo** [comment]
Thanks! Yeah, i am aware of it, and this is on my TODO list. Unfortunately work is has much bigger priority on my TODO list. :(

**Comment by WonderCsabo** [comment]
Well, i tried to upgrade Robolectric, but i ran into a new bug: https://github.com/robolectric/robolectric/issues/2562 .

**Comment by dodgex** [comment]
that sucks :(

**Comment by WonderCsabo** [comment]
@dodgex ready to be reviewed finally!

File: AndroidAnnotations/androidannotations-core/androidannotations-test/pom.xml [hunk 2]
```diff
@@ -62,6 +62,16 @@
 			<artifactId>robolectric</artifactId>
 		</dependency>
 		<dependency>
+			<groupId>org.robolectric</groupId>
```
Comment (dodgex) [inline AndroidAnnotations/androidannotations-core/androidannotations-test/pom.xml]: This file is missing license header update.

File: AndroidAnnotations/androidannotations-core/androidannotations-test/src/test/java/org/androidannotations/test/AwaitingResultFragmentTest.java [hunk 3]
```diff
@@ -45,7 +46,7 @@ public void setUp() {
 	public void testOnResultCalledInFragment() {
 		FragmentStartedActivity_.intent(fragment).startForResult(AwaitingResultFragment.FIRST_REQUEST);
 
-		ShadowActivity a = Robolectric.shadowOf_(fragment.getActivity());
+		ShadowActivity a = (ShadowActivity) ShadowExtractor.extract(fragment.getActivity());
```
Comment (dodgex) [inline AndroidAnnotations/androidannotations-core/androidannotations-test/src/test/java/org/androidannotations/test/AwaitingResultFragmentTest.java]: In other places you static import extract. I think we should be consistent?

File: AndroidAnnotations/androidannotations-core/androidannotations-test/src/test/resources/robolectric.properties [hunk 4]
```diff
@@ -0,0 +1,18 @@
+#
+# Copyright (C) 2010-2016 eBusiness Information, Excilys Group
```
Comment (dodgex) [inline AndroidAnnotations/androidannotations-core/androidannotations-test/src/test/resources/robolectric.properties]: Looks like a new file. why the excilys copyright?

File: AndroidAnnotations/androidannotations-ormlite/ormlite-test/src/test/resources/robolectric.properties [hunk 5]
```diff
@@ -0,0 +1,18 @@
+#
+# Copyright (C) 2010-2016 eBusiness Information, Excilys Group
```
Comment (dodgex) [inline AndroidAnnotations/androidannotations-ormlite/ormlite-test/src/test/resources/robolectric.properties]: Looks like a new file. why the excilys copyright?

File: AndroidAnnotations/androidannotations-otto/otto-test/src/test/resources/robolectric.properties [hunk 6]
```diff
@@ -0,0 +1,18 @@
+#
+# Copyright (C) 2010-2016 eBusiness Information, Excilys Group
```
Comment (dodgex) [inline AndroidAnnotations/androidannotations-otto/otto-test/src/test/resources/robolectric.properties]: Looks like a new file. why the excilys copyright?

File: AndroidAnnotations/androidannotations-rest-spring/rest-spring-test/src/test/resources/robolectric.properties [hunk 7]
```diff
@@ -0,0 +1,18 @@
+#
+# Copyright (C) 2010-2016 eBusiness Information, Excilys Group
```
Comment (dodgex) [inline AndroidAnnotations/androidannotations-rest-spring/rest-spring-test/src/test/resources/robolectric.properties]: Looks like a new file. why the excilys copyright?

File: AndroidAnnotations/androidannotations-roboguice/roboguice-test/src/test/resources/robolectric.properties [hunk 8]
```diff
@@ -0,0 +1,18 @@
+#
+# Copyright (C) 2010-2016 eBusiness Information, Excilys Group
```
Comment (dodgex) [inline AndroidAnnotations/androidannotations-roboguice/roboguice-test/src/test/resources/robolectric.properties]: Looks like a new file. why the excilys copyright?

**Comment by dodgex** [comment]
Oh come on!

All my recent PR do not modify more than 10 files and you hit me with 55 changed files. how did I deserve that? :D


But I just did a fast scroll-over-review. Added some comments. I'll try to find some time to do a better review :)

File: AndroidAnnotations/androidannotations-core/androidannotations-test/src/main/java/org/androidannotations/test/ActivityWithServices.java [hunk 9]
```diff
@@ -144,9 +146,8 @@
 	// @SystemService
 	// HdmiControlManager hdmiControlManager; // hidden API
 
-	// TODO no support yet in Robolectric
-	// @SystemService
-	// InputManager inputManager;
+	@SystemService
+	InputManager inputManager;
```
Comment (dodgex) [inline AndroidAnnotations/androidannotations-core/androidannotations-test/src/main/java/org/androidannotations/test/ActivityWithServices.java]: I think you have to uncomment the [other end](https://github.com/WonderCsabo/androidannotations/blob/robolectric3/AndroidAnnotations/androidannotations-core/androidannotations-test/src/test/java/org/androidannotations/test/ServiceInjectionTest.java#L67) of this test too?

File: AndroidAnnotations/androidannotations-core/androidannotations-test/src/test/java/org/androidannotations/test/res/ResActivityTest.java [hunk 10]
```diff
@@ -78,20 +75,13 @@ public void xmlResAnimNotNull() {
 		assertThat(activity.fade_in).isNotNull();
 	}
 
-	/**
-	 * Cannot be tested right now, because there is no Robolectric shadow class
-	 * for {@link android.view.animation.AnimationUtils AnimationUtils}.
-	 */
-	// @Test
+
```
Comment (dodgex) [inline AndroidAnnotations/androidannotations-core/androidannotations-test/src/test/java/org/androidannotations/test/res/ResActivityTest.java]: unnecessary empty line ;)

**Comment by dodgex** [comment]
@WonderCsabo two open reviews. but only some small points. good work! :)

File: AndroidAnnotations/androidannotations-core/androidannotations-test/src/test/resources/robolectric.properties (hunk 4 above)
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-core/androidannotations-test/src/test/resources/robolectric.properties]: This files was renamed from `org.robolectric.Config.properties`.

File: AndroidAnnotations/androidannotations-ormlite/ormlite-test/src/test/resources/robolectric.properties (hunk 5 above)
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-ormlite/ormlite-test/src/test/resources/robolectric.properties]: This files was renamed from `org.robolectric.Config.properties`.

File: AndroidAnnotations/androidannotations-otto/otto-test/src/test/resources/robolectric.properties (hunk 6 above)
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-otto/otto-test/src/test/resources/robolectric.properties]: This files was renamed from `org.robolectric.Config.properties`.

File: AndroidAnnotations/androidannotations-rest-spring/rest-spring-test/src/test/resources/robolectric.properties (hunk 7 above)
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-rest-spring/rest-spring-test/src/test/resources/robolectric.properties]: This files was renamed from `org.robolectric.Config.properties`.

File: AndroidAnnotations/androidannotations-roboguice/roboguice-test/src/test/resources/robolectric.properties (hunk 8 above)
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-roboguice/roboguice-test/src/test/resources/robolectric.properties]: This files was renamed from `org.robolectric.Config.properties`.

File: AndroidAnnotations/androidannotations-core/androidannotations-test/src/main/java/org/androidannotations/test/ActivityWithServices.java (hunk 9 above)
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-core/androidannotations-test/src/main/java/org/androidannotations/test/ActivityWithServices.java]: Well it is still `null` in the test, so this was uncommented incorrectly.

File: AndroidAnnotations/androidannotations-core/androidannotations-test/src/test/resources/robolectric.properties (hunk 4 above)
Comment (dodgex) [inline AndroidAnnotations/androidannotations-core/androidannotations-test/src/test/resources/robolectric.properties]: Hmm okay. the original just was without copyright header...

File: AndroidAnnotations/androidannotations-core/androidannotations-test/src/main/java/org/androidannotations/test/ActivityWithServices.java [hunk 11]
```diff
@@ -145,9 +145,8 @@
 	// @SystemService
 	// HdmiControlManager hdmiControlManager; // hidden API
 
-	// TODO no support yet in Robolectric
 	// @SystemService
-	// InputManager inputManager;
```
Comment (dodgex) [inline AndroidAnnotations/androidannotations-core/androidannotations-test/src/main/java/org/androidannotations/test/ActivityWithServices.java]: I think we should revert the whole change here?

**Comment by dodgex** [comment]
One last comment in the `ActivityWithServices` but then I'm happy with this PR.

File: AndroidAnnotations/androidannotations-core/androidannotations-test/src/main/java/org/androidannotations/test/ActivityWithServices.java (hunk 11 above)
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-core/androidannotations-test/src/main/java/org/androidannotations/test/ActivityWithServices.java]: You are right,

**Comment by dodgex** [comment]
Thank you! :)

**Pull request merged**